src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/ngapacts/DcirNgapActExtractor.scala

```diff
@@ -70,12 +71,28 @@ class DcirNgapActExtractor(ngapActsConfig: NgapActConfig) extends DcirExtractor[
     import input.sqlContext.implicits._
 
     {
-      if (ngapActsConfig.acts_categories.isEmpty) {
+      if (ngapActsConfig.actsCategories.isEmpty) {
         input.filter(isInExtractorScope _)
       }
       else {
         input.filter(isInExtractorScope _).filter(isInStudy(codes) _)
       }
     }.flatMap(builder _).distinct()
   }
+
+  def dcirIsInCategory(
+                        categories: List[NgapActClassConfig],
+                        row: Row): Boolean = {
+
+    val ngapKeyLetter : String = row.getAs[String]("PRS_NAT_CB2")
+    val ngapCoefficient : String = row.getAs[Double]("PRS_ACT_CFT").toString
+    val prsNatRef: String = row.getAs[Int]("PRS_NAT_REF").toString
+
+    categories
```

Comment:
The statement is written that hides to the reader the logic of the boolean test. It would be better like the following:

```scala
    categories
      .exists(category =>
        (
            category.ngapKeyLetters.contains(ngapKeyLetter) &&
            category.ngapCoefficients.contains(ngapCoefficient)
        ) || 
        category.ngapPrsNatRefs.contains(prsNatRef)
      )
  }
```